libflux/flux-core/src/semantic/sub.rs

```diff
@@ -151,6 +151,18 @@ pub trait Substitutable {
             *self = new;
         }
     }
+
+    /// Apply a substitution to a type variable.
+    fn apply_cow(&self, sub: &dyn Substituter) -> Cow<'_, Self>
+    where
+        Self: Clone + Sized,
+    {
+        match self.apply_ref(sub) {
+            Some(t) => Cow::Owned(t),
+            None => Cow::Borrowed(self),
+        }
+    }
```

Comment:
Sorry, I'm not seeing where this is getting used? Or what its for.

Reply:
Seems I removed the code that used it. It is a helper that lets you call `apply(_cow)` on a `&` reference and immediately use the result in any method that then only requires a `&` reference.

So if you want to display a type where you have applied all substitutions you can do `typ.apply_cow(&sub).to_string()` instead of  `typ.clone().apply(&sub).to_string()` or `typ.apply_ref(&sub).unwrap_or(&typ).to_string()`

Reply:
Okay. Rust might complain if its not used anywhere, but thanks for that explanation.